src/InnoSetup/Environment.iss

```diff
@@ -401,7 +407,20 @@ begin
 
     CmdLine := CmdLine + ' ' + GitRepository +' "' + IDFPath + '"';
     Log('Cloning IDF: ' + CmdLine);
-    DoCmdlineInstall(CustomMessage('DownloadingEspIdf'), CustomMessage('UsingGitToClone'), CmdLine);
+
+    if DoCmdlineInstall(CustomMessage('DownloadingEspIdf'), CustomMessage('UsingGitToClone'), CmdLine) then begin
+      Log('Git clone succeeded.');
+      CloneSuccess := True;
+    end else begin
+      Log('Git clone failed.');
```

Comment:
This line could never be executed as the failure will raise an exception

Reply:
You can replace DoCmdLineInstall with PerformCmdLineInstall, it won't raise an exception. But you need to raise the exception to abort the installation yourself then.
https://github.com/espressif/idf-installer/blob/e04fe05d50eb842ae4f2abe822adf0cda49ae532/src/InnoSetup/Pages/CmdlinePage.iss#L195-L200

Reply:
I will replace it and then test the `Git clone failed` case